### ISSUE TITLE
test-fuzzy-index fails sometimes

### ISSUE THREAD

**Issue by yaneti** [chergert/libdazzle#8]
test-fuzzy-index only sometimes fails on a Fedora rawhide build
glib2-2.53.4-1.fc27.x86_64

ERROR:../tests/test-fuzzy-index.c:126:test_index_builder_basic: assertion failed: (file == NULL)

I seems to happen randomly, say 1 out of 8 times

```
$ ninja test -j8
[0/1] Running all tests.
 1/17 test-application                        OK       0.14 s
 2/17 test-menu-manager2                      OK       0.03 s
 3/17 test-state-machine                      OK       0.08 s
 4/17 test-binding-group                      OK       0.04 s
 5/17 test-signal-group                       OK       0.03 s
 6/17 test-task-cache                         OK       0.03 s
 7/17 test-heap                               OK       0.17 s
 8/17 test-suggestion-buffer                  OK       0.07 s
 9/17 test-fuzzy-index                        FAIL     0.20 s
10/17 test-shortcut-chord                     OK       0.02 s
11/17 test-shortcut-overlays                  OK       0.12 s
12/17 test-shortcut-theme                     OK       0.14 s
13/17 test-trie                               OK       0.76 s
14/17 test-levenshtein                        OK       0.02 s
15/17 test-int-pair                           OK       0.00 s
16/17 test-ring                               OK       0.02 s
17/17 test-list-store                         OK       0.07 s

OK:        16
FAIL:       1
SKIP:       0
TIMEOUT:    0


The output from the failed tests:

 9/17 test-fuzzy-index                        FAIL     0.20 s

--- command ---
G_TEST_SRCDIR='/home/yaneti/cvs/gnome/libdazzle/tests' G_TEST_BUILDDIR='/home/yaneti/cvs/gnome/libdazzle/build/tests' G_DEBUG='gc-friendly' GSETTINGS_BACKEND='memory' PYTHONDONTWRITEBYTECODE='yes' MALLOC_CHECK_='2' /home/yaneti/cvs/gnome/libdazzle/build/tests/test-fuzzy-index
--- stdout ---
/Dazzle/Fuzzy/IndexBuilder/basic: 
--- stderr ---
**
ERROR:../tests/test-fuzzy-index.c:126:test_index_builder_basic: assertion failed: (file == NULL)
-------

Full log written to /home/yaneti/cvs/gnome/libdazzle/build/meson-logs/testlog.txt
FAILED: test 
/usr/bin/python3 -u /usr/bin/mesontest --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.

```

**Comment by yaneti** [comment]
Fixed by cb3851e3665f4c09871faf923ac29181839cd778

**Comment by yaneti** [comment]
The other subtest in test-fuzzy-index also seems to suffer the same. Reopening

**Comment by chergert** [comment]
This was fixed by commit 689fea3a5a3cfe175657d736477d8b7f9119651d and verifies on the architectures that were more prone (racy) to cause the issue.
